src/tests/ftest/launch.py

```diff
@@ -2121,41 +2122,43 @@ def stop_daos_agent_services(test_file, args):
     """Stop any daos_agent.service running on the hosts running servers.
 
     Args:
-        test_file (str): the test python file
+        test_file (dict): a dictionary of the test script/yaml file
         args (argparse.Namespace): command line arguments for this program
 
     Returns:
         int: status code: 0 = success, 512 = failure
 
     """
     service = "daos_agent.service"
-    print("-" * 80)
-    print("Verifying {} after running '{}'".format(service, test_file))
-    if args.test_clients:
-        hosts = list(args.test_clients)
-    else:
-        hosts = list(args.test_servers)
+    client_hosts = get_hosts_from_yaml(test_file["yaml"], args, YAML_KEYS["test_clients"])
     local_host = socket.gethostname().split(".")[0]
-    if local_host not in hosts:
-        hosts.append(local_host)
-    return stop_service(hosts, service)
+    if local_host not in client_hosts:
+        client_hosts.append(local_host)
+    print("-" * 80)
+    print(
+        "Verifying {} on {} after running '{}'".format(
+            service, NodeSet.fromlist(client_hosts), test_file["py"]))
+    return stop_service(client_hosts, service)
 
 
 def stop_daos_server_service(test_file, args):
     """Stop any daos_server.service running on the hosts running servers.
 
     Args:
-        test_file (str): the test python file
+        test_file (dict): a dictionary of the test script/yaml file
         args (argparse.Namespace): command line arguments for this program
 
     Returns:
         int: status code: 0 = success, 512 = failure
 
     """
     service = "daos_server.service"
+    server_hosts = get_hosts_from_yaml(test_file["yaml"], args, YAML_KEYS["test_servers"])
     print("-" * 80)
-    print("Verifying {} after running '{}'".format(service, test_file))
-    return stop_service(list(args.test_servers), service)
+    print(
+        "Verifying {} on {} after running '{}'".format(
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string